### PR TITLE
fix: checking of arguments

### PR DESCRIPTION
**Pull request by Jakub-Vacek**
## Description
Fixes: https://github.com/superfaceai/cli/issues/340
Also fixes other commands with two positional arguments where one missing argument can lead to not relevant error message

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have updated the documentation accordingly. For updating Oclif commands documentation use [oclif-dev](https://github.com/oclif/dev-cli#oclif-dev-readme).
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I haven't repeated the code. (DRY)
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by kysely — COMMENTED** [review]
Is it possible in OCLIF to trigger printout of the USAGE part of the command help msg?
Screenshot 2023-07-25 at 16 08 28

**Comment by Jakub-Vacek** [comment]
> Is it possible in OCLIF to trigger printout of the USAGE part of the command help msg?
> Screenshot 2023-07-25 at 16 08 28

I think there is no easy way now. I tried to add function to our Command base class but all of relevant data lives on static side of classes and is accessible after parse. So I would need to prepare some function that takes instance of Command, picks data (args, descriptions etc.) and prints them to string.

File: src/commands/map.ts [hunk 1]
```diff
@@ -98,6 +98,13 @@ export default class Map extends Command {
     const ux = UX.create();
     const { providerName, profileId, language } = args;
 
+    if (providerName === undefined || profileId === undefined) {
+      throw userError(
+        'Missing provider name or profile id. Please provide them as first and second argument.',
```
Comment (kysely) [inline src/commands/map.ts:103]:
```suggestion
        'Missing provider name or profile ID. Usage: `superface map PROVIDERNAME [PROFILEID]`',
```

File: src/commands/new.ts [hunk 2]
```diff
@@ -67,6 +67,13 @@ export default class New extends Command {
 
     ux.start('Checking input arguments');
 
+    if (providerName === undefined || prompt === undefined) {
+      throw userError(
+        'Missing provider name or prompt. Please provide them as first and second argument.',
```
Comment (kysely) [inline src/commands/new.ts:72]:
```suggestion
        'Missing provider name or prompt. Usage: `superface new PROVIDERNAME [PROMPT]`',
```

File: src/commands/execute.ts [hunk 3]
```diff
@@ -76,6 +76,13 @@ export default class Execute extends Command {
     const ux = UX.create();
     const { providerName, profileId, language } = args;
 
+    if (providerName === undefined || profileId === undefined) {
+      throw userError(
+        'Missing provider name or profile id. Please provide them as first and second argument.',
```
Comment (kysely) [inline src/commands/execute.ts:81]:
```suggestion
        'Missing provider name or profile ID. Usage: `superface execute PROVIDERNAME PROFILEID`',
```

**Review by kysely — APPROVED** [review]
Since an appropriate “help” for each command cannot be triggered after incorrect use, I'd simply put the usage hint directly in the error message. Left suggestions, otherwise looks good 👍

**Review by kysely — APPROVED** [review]

**Pull request merged**